### PR TITLE
Added Facefish rootkit analyzer.

### PR DESCRIPTION
**Pull request by keithjjones**
A protocol analyzer for the Facefish Linux rootkit, as described here:

* https://blog.netlab.360.com/ssh_stealer_facefish_en/
* https://blogs.juniper.net/en-us/threat-research/linux-servers-hijacked-to-implant-ssh-backdoor
* https://thehackernews.com/2021/05/researchers-warn-of-facefish-backdoor.html
* https://securityaffairs.co/wordpress/118388/malware/facefish-backdoor.html

More information and example output in the readme.md in the facefish_rootkit directory.

### CODE REVIEW

**Comment by keithjjones** [comment]
I can't find the reason the tests are failing.  The log the CI refers to is something I can't see.  FWIW the tests don't fail on my mac:

```
$ sudo zkg test .
/[REDACTED]/Source/spicy-analyzers: all tests passed
```

**Comment by keithjjones** [comment]
Tests are still passing for me.  Not sure what is up.  That should be the last of my pushes outside whatever might be failing i can't see in the CI pipeline.

```
$ sudo zkg test .
/[REDACTED]/Source/spicy-analyzers: all tests passed
```

**Comment by rsmmr** [comment]
As this is more specialized than the other analyzers, how about this:  let's merge this into `spicy-analyzers` for now so that it becomes available to folks for installation. Later, once we have an easy way to package Spicy analyzers as standalone zkg packages, we'll split it out into its own package. 

@ckreibich: this looks like a good first candidate for your zkg Spicy template

**Comment by bbannier** [comment]
> Tests are still passing for me. Not sure what is up. That should be the last of my pushes outside whatever might be failing i can't see in the CI pipeline.

@keithjjones, we currently emit the full error message in a separate CI step. You can view it by expanding the GH task details and then _view more details on Cirrus CI_. There locate the _Run stderr_ task (always runs last), e.g.,  for the errors here:

```
[#1] protocol.facefish_rootkit.facefish_generated_pcap [zkg] ... failed
  % 'zeek -C -r ${TRACES}/facefish_rootkit_generated.pcap /Users/admin/.zkg/testing/cirrus-ci-build/clones/cirrus-ci-build/tests/.tmp/protocol.facefish_rootkit.facefish_generated_pcap/facefish_generated_pcap.zeek' failed unexpectedly (exit code 1)
  % cat .stderr
  fatal error in /Users/admin/.zkg/testing/cirrus-ci-build/clones/cirrus-ci-build/tests/.tmp/protocol.facefish_rootkit.facefish_generated_pcap/facefish_generated_pcap.zeek, line 9: can't find spicy-analyzers
```

**Comment by keithjjones** [comment]
@rsmmr sure.  I don't know how to make standalone spicy packages.

@bbannier why can't it find spicy-analyzers?  That's itself.

```  
fatal error in /Users/admin/.zkg/testing/cirrus-ci-build/clones/cirrus-ci-build/tests/.tmp/protocol.facefish_rootkit.facefish_generated_pcap/facefish_generated_pcap.zeek, line 9: can't find spicy-analyzers
```

**Comment by rsmmr** [comment]
> @rsmmr sure. I don't know how to make standalone spicy packages.

Nobody does, that's why I pinged @ckreibich, he is closest to figuring it out. :-)

File: README.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Currently, the following analyzers are included:
 
 - DHCP <sup>[1]</sup>
 - DNS <sup>[1]</sup>
+- [Facefish Rootkit](https://blog.netlab.360.com/ssh_stealer_facefish_en/)
```
Comment (bbannier) [inline README.md:12]:
```suggestion
- [Facefish Rootkit](https://blog.netlab.360.com/ssh_stealer_facefish_en/)
```

File: analyzer/protocol/facefish_rootkit/Readme.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Facefish_Rootkit
+
+This Spicy analyzer will detect the Facefish Linux rootkit C2 as described in:
+
+- https://blog.netlab.360.com/ssh_stealer_facefish_en/
+- https://blogs.juniper.net/en-us/threat-research/linux-servers-hijacked-to-implant-ssh-backdoor
+- https://thehackernews.com/2021/05/researchers-warn-of-facefish-backdoor.html
+- https://securityaffairs.co/wordpress/118388/malware/facefish-backdoor.html
+
+Any detections will be found in "facefish_rootkit.log" while a "Facefish_Rootkit::FACEFISH_ROOTKIT_C2" notice
```
Comment (bbannier) [inline analyzer/protocol/facefish_rootkit/Readme.md:10]:
```suggestion
Any detections will be found in "facefish_rootkit.log" while a "Facefish_Rootkit::FACEFISH_ROOTKIT_C2" notice
```

File: analyzer/protocol/facefish_rootkit/Readme.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+There is a PCAP of the C2 traffic available here:
+
+https://www.joesandbox.com/analysis/355141/0/html#network
+
+The output is as follows:
+
+```
+$ zeek -Cr dump-38fb322cc6d09a6ab85784ede56bc5a7.pcap spicy-analyzers
+
+$ cat conn.log
```
Comment (bbannier) [inline analyzer/protocol/facefish_rootkit/Readme.md:28]:
```suggestion
$ cat conn.log
```

File: analyzer/protocol/facefish_rootkit/Readme.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+1613702572.190619	CCo5Jp4eSxYMobOp3i	192.168.2.20	43450	176.111.174.26	443	tcp	-	3.003396	0	0	S0	-	-	0	S	3	180	0	0	-
+1613702579.089432	CbPIFh4Olo8i1G0KV2	192.168.2.20	43448	176.111.174.26	443	tcp	-	-	-	-	S0	-	-	0	S	1	60	0	0	-
+1613702579.201449	CPyuMa4IZGRzzBEOvh	192.168.2.20	43450	176.111.174.26	443	tcp	-	-	-	-	S0	-	-	0	S	1	60	0	0	-
+1613702587.104275	C7nx2B1vfFr2VroO59	192.168.2.20	43448	176.111.174.26	443	tcp	-	-	-	-	S0	-	-	0	S	1	60	0	0	-
+1613702603.150049	CUcCUW2x5WZS1yWyci	192.168.2.20	43448	176.111.174.26	443	tcp	-	-	-	-	S0	-	-	0	S	1	60	0	0	-
+1613702635.209441	ChmbMq44oJeZIR6tVf	192.168.2.20	43448	176.111.174.26	443	tcp	-	-	-	-	S0	-	-	0	S	1	60	0	0	-
+1613702587.216303	Clfyzh4ifzsqqUzvqc	192.168.2.20	43450	176.111.174.26	443	tcp	spicy_facefish_rootkit	29.625531	4304	32	S1	-	-	0	ShADTad	19	5348	12	688	-
+#close	2021-06-03-14-53-30
+
+$ cat facefish_rootkit.log
```
Comment (bbannier) [inline analyzer/protocol/facefish_rootkit/Readme.md:47]:
```suggestion
$ cat facefish_rootkit.log
```

**Comment by keithjjones** [comment]
Holy cow the tests pass.  Thanks for the help @bbannier !  This should be ready for merge now.

**Review by bbannier — APPROVED** [review]
This looks good to me, left only trivial nits (this more whitespace stuff in here which I didn't flag since we are inconsistent about this elsewhere as well, e.g., newlines at the end of files).

File: analyzer/protocol/facefish_rootkit/Readme.md (hunk 2 above)
Comment (keithjjones) [reply analyzer/protocol/facefish_rootkit/Readme.md:10]: This is actually a natural space in the sentence.

File: analyzer/protocol/facefish_rootkit/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2021 by the Zeek Project. See LICENSE for details.
+
+spicy_add_analyzer(Facefish_Rootkit facefish_rootkit.spicy facefish_rootkit_zeek.spicy facefish_rootkit.evt)
+
```
[the remark this answers is not in the document]
Comment (keithjjones) [reply analyzer/protocol/facefish_rootkit/CMakeLists.txt]: Fixed

**Comment by keithjjones** [comment]
The macos test failure is homebrew failing.

**Pull request merged**